Foundation/NSObjCRuntime.swift

```diff
@@ -116,24 +116,21 @@ internal func _NSGetSizeAndAlignment(type: _NSSimpleObjCType,
 }
 
 public func NSGetSizeAndAlignment(typePtr: UnsafePointer<Int8>,
-                                  _ sizep: UnsafeMutablePointer<Int>,
-                                  _ alignp: UnsafeMutablePointer<Int>) -> UnsafePointer<Int8> {
+                                  _ sizep: UnsafeMutablePointer<Int>?,
+                                  _ alignp: UnsafeMutablePointer<Int>?) -> UnsafePointer<Int8> {
     let type = _NSSimpleObjCType(UInt8(typePtr.pointee))!
 
     var size : Int = 0
     var align : Int = 0
     
     if !_NSGetSizeAndAlignment(type, &size, &align) {
-        return nil
-    }
-    
-    if sizep != nil {
-        sizep.pointee = size
+        // FIXME: This used to return nil, but the corresponding Darwin
+        // implementation is defined as returning a non-optional value.
+        fatalError("invalid type encoding")
```

Comment:
Here's a place where the Darwin API disagrees with the open-source implementation. The one call I saw, though, didn't even check the result.